inc/class-yoast-acf-analysis-requirements.php

```diff
@@ -6,69 +6,40 @@
  */
 class Yoast_ACF_Analysis_Requirements {
 
-	const MIN_WPSEO_VERSION = 3.2;
+	/** @var Yoast_ACF_Analysis_Dependency[] List of registered dependencies. */
+	protected $dependencies = array();
 
 	/**
-	 * Add notifications to admin if plugins ACF or WordPress SEO are not present, deactivates plugin if so.
+	 * Adds a dependency.
 	 *
-	 * @return boolean If all requirements for this plugin are met
-	 */
-	public function check() {
-
-		// Require ACF and Yoast.
-		if ( current_user_can( 'activate_plugins' ) ) {
-			$deactivate = false;
-
-			// ACF.
-			if ( ! is_plugin_active( 'advanced-custom-fields/acf.php' ) && ! is_plugin_active( 'advanced-custom-fields-pro/acf.php' ) ) {
-				add_action( 'admin_notices', array( $this, 'acf_requirements_not_met' ) );
-				$deactivate = true;
-			}
-
-			// Yoast SEO for WordPress.
-			if ( ! is_plugin_active( 'wordpress-seo/wp-seo.php' ) && ! is_plugin_active( 'wordpress-seo-premium/wp-seo-premium.php' ) ) {
-				add_action( 'admin_notices', array( $this, 'yoast_seo_requirements_not_met' ) );
-				$deactivate = true;
-			} else {
-				if ( defined( 'WPSEO_VERSION' ) ) {
-					if ( version_compare( substr( WPSEO_VERSION, 0, 3 ), self::MIN_WPSEO_VERSION, '<' ) ) {
-						add_action( 'admin_notices', array( $this, 'yoast_seo_requirements_not_met' ) );
-						$deactivate = true;
-					}
-				}
-			}
-
-			// Deactivate when we cannot do the job we are hired to do.
-			if ( $deactivate ) {
-				deactivate_plugins( plugin_basename( YOAST_ACF_ANALYSIS_FILE ) );
```

Comment:
When I was testing the functionality, I noticed that the message would only be there for one page-load. I understand the concept of deactivating the plugin, but a user will generally assume everything went ok when the message say 'Plugin successfully activated'. Moving on with the next task they have, it's very easy to miss it or to forget about it shortly after.

Having that warning on the screens that functionality is not as expected, with the proper action to resolve the problem is a big help.

It also reduces the amount of 'actions' a user has to take to get the wanted functionality, one plugin does not have to be activated again after meeting all requirements.
And the error will disappear whenever all requirements are met. This means that if you install Yoast SEO you will still see the message that ACF needs to be installed as well, you don't need to activate the glue plugin again.

tldr;
Having errors in your face is not a bad thing, they will help the user to achieve the goal they have. It also reduces the number of actions they have to do to reach the goal.